apps/opik-backend/src/main/java/com/comet/opik/domain/DatasetDAO.java

```diff
@@ -65,6 +65,46 @@ int update(@Bind("workspace_id") String workspaceId,
     long findCount(@Bind("workspace_id") String workspaceId, @Define("name") @Bind("name") String name,
             @Define("with_experiments_only") boolean withExperimentsOnly);
 
+    @SqlQuery("SELECT COUNT(*) FROM datasets " +
+            "WHERE workspace_id = :workspace_id " +
+            "AND id IN (<ids>) " +
+            "<if(name)> AND name like concat('%', :name, '%') <endif> ")
+    @UseStringTemplateEngine
+    @AllowUnusedBindings
+    long findCountByIds(@Bind("workspace_id") String workspaceId, @BindList("ids") Set<UUID> ids,
+            @Define("name") @Bind("name") String name);
+
+    @SqlQuery("SELECT * FROM datasets " +
+            "WHERE workspace_id = :workspace_id " +
+            "AND id IN (<ids>) " +
+            "<if(name)> AND name like concat('%', :name, '%') <endif> " +
+            " ORDER BY id DESC " +
+            " LIMIT :limit OFFSET :offset ")
+    @UseStringTemplateEngine
+    @AllowUnusedBindings
+    List<Dataset> findByIds(@Bind("workspace_id") String workspaceId, @BindList("ids") Set<UUID> ids,
+            @Define("name") @Bind("name") String name, @Bind("limit") int limit, @Bind("offset") int offset);
+
+    @SqlQuery("SELECT COUNT(*) FROM datasets " +
+            "WHERE workspace_id = :workspace_id " +
+            "AND id IN (SELECT id FROM experiment_dataset_ids_<table_name>) " +
+            "<if(name)> AND name like concat('%', :name, '%') <endif> ")
+    @UseStringTemplateEngine
+    @AllowUnusedBindings
+    long findCountByTempTable(@Bind("workspace_id") String workspaceId, @Define("table_name") String tableName,
+            @Define("name") @Bind("name") String name);
+
+    @SqlQuery("SELECT * FROM datasets " +
+            "WHERE workspace_id = :workspace_id " +
+            "AND id IN (SELECT id FROM experiment_dataset_ids_<table_name>) " +
+            "<if(name)> AND name like concat('%', :name, '%') <endif> " +
+            " ORDER BY id DESC " +
+            " LIMIT :limit OFFSET :offset ")
+    @UseStringTemplateEngine
+    @AllowUnusedBindings
+    List<Dataset> findByTempTable(@Bind("workspace_id") String workspaceId, @Define("table_name") String tableName,
```

Comment:
The usage of temporary tables for this whole retrieval is concerning. I don't think you get much benefit, as looking at the service you don't require persistence accross calls/sessions. On the other hand, you get many downsides from managing this, specially under error scenarios, such as leaving abandoned tables. 

I believe it's simply better and easier to perform this search operation on memory.

Reply:
The main point for the temp table is MySQL query string limit; in the past, we had a similar error in Clickhouse; if the query is too big, the database will reject it. I tested just 5000, and it worked, so I used it as default. However, I didn't check how much it would support. That is point, for the temp table

Reply:
The main issue here is the MySQL max query string size. If we have, for some reason, thousands of dataset IDs,  and it goes over the limit, we will get an error. I set 5000 to be safe and tested locally, and it worked. But similar to the error we got once from Clickhouse, we can also exceed such limit in MySQL. That is the main point for the temp table